### PR TITLE
Dropbox api upgrade

### PR DESCRIPTION
**Pull request by patroqueeet**
adressing #301

tests not proven as #227#issuecomment-298834277

but backup worked for me

### CODE REVIEW

**Comment by jschneier** [comment]
Thanks, glad to hear you got it working. I think I'm going to remove the progress bar. Also things like `xrange` are Python 2.x only but this looks okay - I can clean up the testing if you'd like.

**Comment by patroqueeet** [comment]
I added one test according to the other pattern you used.

for the progress bar: maybe based on setting? reason I added it was, that I'm compressing the files to upload and it's about 2.5GB for me. the uplink to dropbox makes 19MB/s so 2.5GB will take some time. to calm the user it felt good to have some signs of life showing :) - your decision :)

let me know if there are any questions!

**Comment by zuck** [comment]
Tic-toc...June 28, 2017 is very close now...Merging this PR is VERY urgent!!

File: storages/backends/dropbox.py [hunk 1]
```diff
@@ -59,6 +94,31 @@ def __init__(self, oauth2_access_token=None, root_path=None):
                                        "'settings.DROPBOX_OAUTH2_TOKEN'.")
         self.client = Dropbox(oauth2_access_token)
 
+    def _chunked_upload(self, content, dest_path):
+        """use chunked upload session for large files. stolen from
+        https://goo.gl/4XV0yT"""
```
Comment (jleclanche) [inline storages/backends/dropbox.py]: Use the full URL: https://stackoverflow.com/documentation/dropbox-api/409/

**Comment by scps950707** [comment]
any news?

**Comment by zuck** [comment]
Deadline moved to September 28th, 2017:

https://blogs.dropbox.com/developers/2017/06/updated-api-v1-deprecation-timeline/

We have a little more time, but this issue has to be fixed ASAP, please!

**Comment by jschneier** [comment]
@scps950707 @zuck the API upgrade is done (in master), the only thing missing from this PR is the large files upload.

**Comment by jschneier** [comment]
All of these are now fixed in master.

**Pull request closed without merging**